proxy/src/control/observe.rs

```diff
@@ -14,7 +15,7 @@ use telemetry::tap::{Tap, Taps};
 
 #[derive(Clone, Debug)]
 pub struct Observe {
-    next_id: usize,
+    next_id: Arc<AtomicUsize>,
```

Comment:
If we *really* wanted to allocate less, the `next_id` and the `taps` could be within the same `Arc`... If we don't construct too many `Observe`s, then it wouldn't matter.

Reply:
the `taps` arc is shared externally, and i don't expect too many Observe instances